agents/text_2_sql_bedrock_agent/dependencies/build_infrastructure.py

```diff
@@ -0,0 +1,467 @@
+from config import *
+
+
+# ### Create S3 bucket and upload API Schema
+# 
+# Agents require an API Schema stored on s3. Let's create an S3 bucket to store the file and upload the file to the newly created bucket
+
+
+
+# Create S3 bucket for Open API schema
+s3bucket = s3_client.create_bucket(
+    Bucket=bucket_name
+    #CreateBucketConfiguration={ 'LocationConstraint': region } 
```

Comment:
can you add a code to detect the region before and decide if the location constraint is required?